cache/lru_cache.cc

```diff
@@ -505,10 +505,16 @@ uint32_t LRUCache::GetHash(Handle* handle) const {
 
 void LRUCache::DisownData() {
 // Do not drop data if compile with ASAN to suppress leak warning.
+#if defined(__clang__)
+#if !defined(__has_feature) || !__has_feature(address_sanitizer)
+  shards_ = nullptr;
```

Comment:
@taewookoh Is there any particular reason you remove `num_shards_ = 0` from the clang case?